src/Controls/src/Core/AppThemeBinding.cs

```diff
@@ -8,10 +8,38 @@ namespace Microsoft.Maui.Controls
 {
 	class AppThemeBinding : BindingBase
 	{
+		class AppThemeProxy : Element
+		{
+			public AppThemeProxy(Element parent, AppThemeBinding binding)
+			{
+				_parent = parent;
+				Binding = binding;
+				this.SetDynamicResource(AppThemeProperty, "__ApplicationTheme__");
+				((IElementDefinition)parent).AddResourcesChangedListener(OnParentResourcesChanged);
+			}
+
+			public static BindableProperty AppThemeProperty = BindableProperty.Create("AppTheme", typeof(AppTheme), typeof(AppThemeBinding), AppTheme.Unspecified,
```

Comment:
Will it be more expensive to have a separate proxy Element object with a single extra property vs just adding this to maybe the base Element?